atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraTimestampsEteTest.java

```diff
@@ -0,0 +1,103 @@
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.awaitility.Awaitility;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.ete.cassandra.util.CassandraCommands;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.ValueType;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoResource;
+import com.palantir.atlasdb.todo.generated.TodoTable;
+
+public class CassandraTimestampsEteTest {
+    private static final Todo TODO = ImmutableTodo.of("todo");
+    private static final Todo TODO_2 = ImmutableTodo.of("todo_two");
+    private static final String CASSANDRA_CONTAINER_NAME = "cassandra";
+    private static final long ID = 1L;
+
+    private TodoResource todoClient = EteSetup.createClientToSingleNode(TodoResource.class);
+
+    @Before
+    public void setup() {
+        todoClient.truncate();
+    }
+
+    @After
+    public void cleanup() {
+        todoClient.truncate();
+    }
+
+    @Test
+    public void timestampsInOneSSTableAreCurrent() throws IOException, InterruptedException {
+        long firstWriteTimestamp = todoClient.addTodoWithIdAndReturnTimestamp(ID, TODO);
+        todoClient.addTodoWithIdAndReturnTimestamp(ID, TODO_2);
+
+        sweepUntilNoValueExistsAtTimestamp(ID, firstWriteTimestamp);
+
+        CassandraCommands.nodetoolFlush(CASSANDRA_CONTAINER_NAME);
+
+        List<String> ssTables = CassandraCommands.nodetoolGetSSTables(
+                CASSANDRA_CONTAINER_NAME,
+                "atlasete",
+                TableReference.create(Namespace.DEFAULT_NAMESPACE, TodoTable.getRawTableName()),
+                ValueType.FIXED_LONG.convertFromJava(ID));
+        String sstableMetadata = CassandraCommands.ssTableMetadata(
+                CASSANDRA_CONTAINER_NAME, Iterables.getOnlyElement(ssTables));
+
+        assertMinimumTimestampIsAtLeast(sstableMetadata, firstWriteTimestamp);
+        assertDroppableTombstoneRatioPositive(sstableMetadata);
+    }
+
+    private void sweepUntilNoValueExistsAtTimestamp(long id, long timestamp) {
+        Awaitility.waitAtMost(30, TimeUnit.SECONDS)
+                .pollInterval(1, TimeUnit.SECONDS)
+                .until(() -> {
+                        todoClient.runIterationOfTargetedSweep();
+                        return todoClient.doesNotExistBeforeTimestamp(id, timestamp);
+                });
+    }
+
+    private void assertMinimumTimestampIsAtLeast(String sstableMetadata, long expectedMinimum) {
+        Pattern pattern = Pattern.compile("Minimum timestamp: ([0-9]*)");
+        Matcher timestampMatcher = pattern.matcher(sstableMetadata);
+        assertThat(timestampMatcher.find()).isTrue();
```

Comment:
Does this give a friendly error message when it fails?

Reply:
No (it gives `expected: <true> but was: <false>`) but I've used assertj's `as` to give a more friendly error message (`[SSTableMetadata output contains the minimum timestamp] expected: <true> but was: <false>`). I did the same for the other regex match as well.